### DISCUSSION TITLE
Quality of image when using watermark

### DISCUSSION THREAD

**Discussion by vgabunia** [imgproxy/imgproxy#1310] [Q&A] [answered]
Hi, 
When I'm try to use watermarks my image becomes blue like.

example:
https://image.liontrans.com/insecure/w:1600/h:1200/wm:0.5:soea:0:0:0.2/f:webp/plain/https://liontransblobstorage.blob.core.windows.net/lot-images/ProductId-6444/10fcf458fb924c1ba855a9b0bdb4e4a3.jpg
what I'm doing wrong
![LION-WMs](https://github.com/user-attachments/assets/d5fc22d6-fcb6-4061-ae12-57e086ae8fd6)

**Comment by DarthSim** [comment]
Hey @vgabunia!

Can I take a look at the watermark file?

**Reply by vgabunia** [reply]
Hi, sure.
https://s3.filebin.net/filebin/4d08dd472ecf384c319b28381241a49a5431980efa0fc3270e64f1affe150e48/1c84b9e3e6adbc9f619310014c9a9c2ccd868332a22caf8304524a4f0ec4cd8f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=7pMj6hGeoKewqmMQILjm%2F20240729%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240729T124357Z&X-Amz-Expires=60&X-Amz-SignedHeaders=host&response-cache-control=max-age%3D60&response-content-disposition=filename%3D%22lionauction-logo__2___2_.svg%22&response-content-type=image%2Fsvg%2Bxml&X-Amz-Signature=cd2a3c0d9620c1252e7e5cc61582cbb24162b71242100fc1c4de3b71f4beafe7

**Reply by vgabunia** [reply]
> Hey @vgabunia!
> 
> Can I take a look at the watermark file?

Hi,
I do something wrong?

**Reply by DarthSim** [reply]
Sorry for the delay. Could you attach the watermark right here?

**Reply by vgabunia** [reply]
> Sorry for the delay. Could you attach the watermark right here?
![Uploading Mediamodifier-Design.svg…]()

**Comment by vgabunia** [comment]
![Mediamodifier-Design](https://github.com/user-attachments/assets/3db55a8c-7f2b-475a-ae41-73afef7dc8d8)

**Reply by vgabunia** [reply]
Hi,
Sure, that is SVG file

**Reply by DarthSim** [reply]
Which version of imgproxy do you use? I can't reproduce the issue with the latest version

**Reply by vgabunia** [reply]
Hi,
version is 3.25.0 - current one
I getting same result when I use default URL as watermark (commented in my .env file)
![image](https://github.com/user-attachments/assets/2464f06d-f8a1-4f9c-ac48-954c6f3ef53a)

**Reply by DarthSim** [reply]
Do you use our Docker image or build imgproxy from the source?

**Reply by vgabunia** [reply]
> Do you use our Docker image or build imgproxy from the source?

Hi, thank you for response.
I Using direct installation in My Virtual machine with ubuntu OS.

**Reply by DarthSim** [reply]
Unfortunately, we can't guarantee fully proper work of imgproxy built from the source. imgproxy has a ton of dependencies, and a bug in any of them may lead to incorrect work of imgproxy. And since I can't reproduce the issue using dockerized imgproxy, I believe, this is the case. In our Docker image, we keep all the dependencies updated and patched (in case a bug is not fixed yet in the upstream).

**Reply by vgabunia** [reply]
> Unfortunately, we can't guarantee fully proper work of imgproxy built from the source. imgproxy has a ton of dependencies, and a bug in any of them may lead to incorrect work of imgproxy. And since I can't reproduce the issue using dockerized imgproxy, I believe, this is the case. In our Docker image, we keep all the dependencies updated and patched (in case a bug is not fixed yet in the upstream).

Hi,
thank you for Reply.
Where I Can see dependency list? 
If i understand correctly, you Recommend use Docker version of Software Correct?

**Reply by DarthSim** [reply, marked as answer]
imgproxy itself has a single first-level dependency: https://github.com/libvips/libvips, but libvips also has lots of dependencies. So yes, I **highly** recommend using our Docker image instead of building imgproxy from source.

**Reply by vgabunia** [reply]
Hi,
You absolutely right
https://image.liontrans.com/insecure/wm:0.5:soea:0:0:0.2/plain/https://liontransblobstorage.blob.core.windows.net/lot-images/ProductId-6445/9874800767bc45ce9544005a85a3559a.jpg
now works well.
thank you for right recomendations.
